easybuild/easyconfigs/j/jax/jax-0.2.19-fosscuda-2020b.eb

```diff
@@ -0,0 +1,117 @@
+# This file is an EasyBuild reciPY as per https://github.com/easybuilders/easybuild
+# Author: Denis Kristak
+easyblock = 'PythonBundle'
+
+name = 'jax'
+version = '0.2.19'
+local_jaxlib_ver = '0.1.70'
+homepage = 'https://pypi.python.org/pypi/jax'
+description = """Composable transformations of Python+NumPy programs: differentiate, vectorize, JIT to GPU/TPU, and
+more"""
+
+toolchain = {'name': 'fosscuda', 'version': '2020b'}
+
+builddependencies = [
+    ('Bazel', '3.7.2'),
+    ('pytest-xdist', '2.3.0'),
+    ('pytest-benchmark', '3.4.1'),
+]
+
+dependencies = [
+    ('Python', '3.8.6'),
+    ('SciPy-bundle', '2020.11'),
+    ('Cython', '0.29.22'),
+    ('flatbuffers-python', '1.12'),
+    ('cuDNN', '8.0.4.30', '-CUDA-%(cudaver)s', True),
+    ('NCCL', '2.8.3', '-CUDA-%(cudaver)s'),
+]
+
+# downloading TensorFlow tarball to avoid that Bazel downloads it during the build
+# note: this *must* be the exact same commit as used in jaxlib-*/WORKSPACE
+local_tf_commit = '4039feeb743bc42cd0a3d8146ce63fc05d23eb8d'
+
+local_tf_builddir = "%%(builddir)s/tensorflow-%s" % local_tf_commit
+
+local_jax_prebuildopts = "sed -i -e 's$pathToSed$%s$g' WORKSPACE && " % local_tf_builddir
+
+local_jax_build_cmd = 'python build/build.py '
+local_jax_build_cmd += '--target_cpu_features=native '
+local_jax_build_cmd += '--bazel_startup_options="--output_user_root=%(builddir)s" '
+local_jax_build_cmd += '--bazel_path="$EBROOTBAZEL/bin/bazel" '
+local_jax_build_cmd += '--bazel_options=--subcommands '
+local_jax_build_cmd += '--bazel_options=--jobs=%(parallel)s --bazel_options=--action_env=PYTHONPATH '
+local_jax_build_cmd += '--bazel_options=--action_env=EBPYTHONPREFIXES '
+local_jax_build_cmd += '--enable_cuda --enable_nccl --cuda_compute_capabilities=%(cuda_compute_capabilities)s '
+local_jax_build_cmd += '--cuda_path=$EBROOTCUDA --cudnn_path=$EBROOTCUDNN '
```

Comment:
@branfosj Together with the stuff we have below, I think we should look into a custom easyblock for `jaxlib`, just to clean this stuff up a bit...

If you agree, I think that's something @deniskristak can look into

Reply:
Yes an easyblock will be needed. We need to control when we set `--target_cpu_features=native `, as this adds `-march=native` and that does not work on POWER.

Reply:
We could use a runtime patch that changes the `-march=native` that gets injected based on which CPU architecture we're on?